activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -638,7 +637,7 @@ def find_last(limit)
       end
 
       def ordered_relation
-        if order_values.empty? && (model.implicit_order_column || !model.query_constraints_list.nil? || primary_key)
+        if order_values.empty? && _order_columns.any?
```

Comment:
personal pet peeve. `any?` isn't the inverse of `empty?`, it ignores falsy members.

Not a huge deal here, but IMO `!_order_columns.empty?` is clearer.

Reply:
I always forget about that caveat. Addressed 👍🏽